src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -155,8 +161,27 @@ ConvSolution InitInvokerFactoryNHWC(const ExecutionContext&,
                         auto zero           = 0.0f;
                         const auto& tensors = data_ctx.tensors;
                         SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
+
+                        void* gpu_buf    = nullptr;
+                        size_t num_bytes = sh_conv_ptr->GetWorkSpaceSize(argument_ptr.get());
+                        if(num_bytes != 0)
+                        {
+                            // allocate gpu workspace memory
+                            HIP_CHECK(hipMalloc(&gpu_buf, num_bytes));
+                            assert(gpu_buf != nullptr);
```

Comment:
This should be `if(...) thow ...` since the following code would likely fail hard if the assertion fails in release. Solver selection mechanism relies on explicit failure to provide a correct solution.

Reply:
updated